File: bulk/go/mass_assign_01.go
// CWE-915: Mass Assignment via JSON decode into struct pointer
package main
import (
	"encoding/json"
	"net/http"
)
type User struct {
	Name    string
	IsAdmin bool
}
func updateUser(w http.ResponseWriter, r *http.Request) {
	var u User
	json.NewDecoder(r.Body).Decode(&u) // IsAdmin settable from request body
}

File: bulk/java/MassAssign01.java
// CWE-915: Mass Assignment via Spring @ModelAttribute
import org.springframework.web.bind.annotation.*;
public class MassAssign01 {
    static class User { public String name; public boolean isAdmin; }
    @PostMapping("/update")
    public User update(@ModelAttribute User user) {
        return user; // isAdmin can be set via form field
    }
}
